apps/web/src/pages/Applications/ApplicationProfilePage/ApplicationProfilePage.tsx

```diff
@@ -75,16 +100,170 @@ export const getPageInfo: GetApplicationPageInfo = ({
+          id: "G1jegJ",
+          description:
+            "Error message displayed when user attempts to submit incomplete profile",
+        }),
+      );
+    }
+
+    return false;
+  };
 
-  return <Heading data-h2-margin-top="base(0)">{pageInfo.title}</Heading>;
+  const handleUpdate: SectionProps["onUpdate"] = (userId, userData) => {
+    return executeUpdateMutation({
+      id: userId,
+      user: userData,
+    }).then((res) => res.data?.updateUserAsUser);
+  };
+
+  const sectionProps = {
+    user,
+    isUpdating,
+    onUpdate: handleUpdate,
+  };
+
+  return (
+    <>
+      <Heading data-h2-margin-top="base(0)">{pageInfo.title}</Heading>
+      <p>
+        {intl.formatMessage({
+          defaultMessage:
+            "This step includes all of your profile information required for this application. Each section has a series of required fields that must be completed to move on to the next step. Completing this profile information will only happen once, and the information you provide here will auto-populate on applications you submit moving forward.",
+          id: "D9Elm9",
+          description: "Application step to complete your profile, description",
+        })}
+      </p>
+      <div
+        data-h2-margin="base(x2, 0)"
+        data-h2-display="base(flex)"
+        data-h2-flex-direction="base(column)"
+        data-h2-gap="base(x1, 0)"
+      >
+        <PersonalInformation {...sectionProps} />
+        <WorkPreferences {...sectionProps} />
+        <DiversityEquityInclusion {...sectionProps} />
+        <GovernmentInformation {...sectionProps} />
+        <LanguageProfile {...sectionProps} />
+      </div>
+      <Separator
+        orientation="horizontal"
+        data-h2-background-color="base(gray.lighter)"
+        data-h2-margin="base(x2, 0)"
+        decorative
+      />
+      <div
+        data-h2-display="base(flex)"
+        data-h2-gap="base(x.25, x.5)"
+        data-h2-flex-wrap="base(wrap)"
+        data-h2-flex-direction="base(column) l-tablet(row)"
+        data-h2-align-items="base(flex-start) l-tablet(center)"
+      >
+        <form onSubmit={handleNavigation}>
+          <Button
+            type="submit"
+            color="primary"
+            mode="solid"
+            disabled={submitting}
+          >
+            {intl.formatMessage({
+              defaultMessage: "Let's go!",
+              id: "r6z4HM",
+              description: "Link text to begin the application process",
+            })}
+          </Button>
+        </form>
+        <Link
+          type="button"
+          mode="inline"
+          color="secondary"
+          href={applicantDashboard ? paths.dashboard() : paths.myProfile()}
+        >
+          {intl.formatMessage({
+            defaultMessage: "Save and quit for now",
+            id: "U86N4g",
+            description: "Action button to save and exit an application",
+          })}
+        </Link>
```

Comment:
This "Save and quit" is supposed to actually save before navigating

![image](https://user-images.githubusercontent.com/40485260/232571955-37a26fcf-c01a-4297-bc72-e3731fbf505f.png)

Reply:
I'm not even sure this is possible since they are all separate forms and not one single form 🤔 This is true for all other steps where the cancel button does not save anything. Honestly, the text for all the cancel buttons is slightly misleading 😢

Reply:
Ah, well, run it by Josh then I suppose so copy and flow can be adjusted if needed. May need to drop mass form updates elsewhere.

Reply:
Might have to reach out to Josh directly.  Tristan's suggestion seems reasonable to me.
> Would it be okay if it just threw an error toast if you still have open forms, telling you to save or cancel before finishing?

Reply:
We've made buttons before which manually triggered the submission of multiple forms, but it's always tricky 😩

Reply:
Should we set aside a bit of time to chat with Josh about this?

Reply:
I think so, these "Save and..." links on the application are very confusing because almost all of them don't save anything right now and seem misleading in most cases...at least that is how everyone has been coding it right now 😅 .

Reply:
I believe I implemented the solution suggested by @substrae properly 😅 When you have an open + dirty form, "Save and quit" shows a toast and then scrolls you to the section. Any action on a form marks it as clean (save, cancel). If all forms are clean, it directs you to either the applicant dashboard or profile page depending on the `APPLICANT_DASHBOARD` feature flag.

- [prevent quitting with dirty forms](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6296/commits/b081cf0c7d02bc368079be5c918ed576bdb85502)
- [move reused code to a hook](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6296/commits/542077e6052c397943247f4e7883ae8ef4413609)

Reply:
This is why I find it confusing 😅. What I saw in your recording is the expected functionality (I think). It is the "save and quit for now" that has the functionality to block the next step which is the part i find confusing because it actually does the opposite of saving and asks that you do it manually or clear out any changes before you can quit 😛